src/Compilers/Core/CodeAnalysisTest/DefaultAnalyzerAssemblyLoaderTests.cs

```diff
@@ -28,642 +37,846 @@ public class AssemblyLoadTestFixtureCollection : ICollectionFixture<AssemblyLoad
-            loader.AddDependencyLocation(_testFixture.AnalyzerReferencesSystemCollectionsImmutable2.Path);
+                loader.AddDependencyLocation(testFixture.UserSystemCollectionsImmutable.Path);
+                loader.AddDependencyLocation(testFixture.AnalyzerReferencesSystemCollectionsImmutable2.Path);
 
-            Assembly analyzerAssembly = loader.LoadFromPath(_testFixture.AnalyzerReferencesSystemCollectionsImmutable2.Path);
-            var analyzer = analyzerAssembly.CreateInstance("Analyzer")!;
-            analyzer.GetType().GetMethod("Method")!.Invoke(analyzer, new object[] { sb });
-            Assert.Equal(ExecutionConditionUtil.IsCoreClr ? "1" : "42", sb.ToString());
+                Assembly analyzerAssembly = loader.LoadFromPath(testFixture.AnalyzerReferencesSystemCollectionsImmutable2.Path);
+                var analyzer = analyzerAssembly.CreateInstance("Analyzer")!;
+                analyzer.GetType().GetMethod("Method")!.Invoke(analyzer, new object[] { sb });
+                Assert.Equal(ExecutionConditionUtil.IsCoreClr ? "1" : "42", sb.ToString());
+            });
         }
 
-        [ConditionalFact(typeof(WindowsOnly), typeof(CoreClrOnly))]
-        public void AssemblyLoading_NativeDependency()
+        [Fact(Skip = "https://github.com/dotnet/roslyn/issues/66104")]
+        public void AssemblyLoading_CompilerDependencyDuplicated()
         {
-            var loader = new DefaultAnalyzerAssemblyLoader();
-            loader.AddDependencyLocation(_testFixture.AnalyzerWithNativeDependency.Path);
+            Run(static (DefaultAnalyzerAssemblyLoader loader, AssemblyLoadTestFixture testFixture) =>
+            {
+                var assembly = typeof(ImmutableArray<int>).Assembly;
+
+                // Copy the dependency to a new location to simulate it being deployed by an 
+                // analyzer / generator
+                using var tempRoot = new TempRoot();
+                var destFile = tempRoot.CreateDirectory().CreateOrOpenFile($"{assembly.GetName().Name}.dll");
+                destFile.CopyContentFrom(assembly.Location);
+                loader.AddDependencyLocation(destFile.Path);
+
+                var copiedAssembly = loader.LoadFromPath(destFile.Path);
+                Assert.Single(AppDomain.CurrentDomain.GetAssemblies().Where(x => x.FullName == assembly.FullName));
+                Assert.Same(copiedAssembly, assembly);
+            });
+        }
 
-            Assembly analyzerAssembly = loader.LoadFromPath(_testFixture.AnalyzerWithNativeDependency.Path);
-            var analyzer = analyzerAssembly.CreateInstance("Class1")!;
-            var result = analyzer.GetType().GetMethod("GetFileAttributes")!.Invoke(analyzer, new[] { _testFixture.AnalyzerWithNativeDependency.Path });
-            Assert.Equal(0, Marshal.GetLastWin32Error());
-            Assert.Equal(FileAttributes.Archive, (FileAttributes)result!);
+        [ConditionalFact(typeof(WindowsOnly))]
```

Comment:
I didn't follow why this test is not longer applicable to Core. Can you clarify?

Reply:
Previously the test only ran on Windows CoreClr, now it runs on Windows CoreClr or Framework. It's an expansion of the test.